WalletWasabi/WabiSabi/Client/ArenaClient.cs

```diff
@@ -94,5 +94,43 @@ public async Task ConfirmConnectionAsync(Guid roundId, Guid aliceId, IEnumerable
 			AmountCredentialClient.HandleResponse(confirmConnectionResponse.ZeroAmountCredentials, zeroAmountCredentialResponseValidation);
 			WeightCredentialClient.HandleResponse(confirmConnectionResponse.ZeroWeightCredentials, zeroWeightCredentialResponseValidation);
 		}
+
+		public async Task SignTransactionAsync(Guid roundId, IEnumerable<ICoin> coinsToSign, BitcoinSecret bitcoinSecret, Transaction unsignedCoinJoin)
```

Comment:
The `ICoin` abstraction is a higher abstraction that works well at the wallet level but we are working in a lower level here. The correct abstraction level here is `OutPoint`, the same we use in `RegisterInputAsync`.

Reply:
Forget it, I said an stupidity. I see we need the coin to sign.